src/elements/form/extras/address/Address.js

```diff
@@ -73,17 +82,25 @@ export class NovoAddressElement implements ControlValueAccessor {
 
     updateControl() {
         this.onModelChange(this.model);
+        this.changeDetectorRef.markForCheck();
     }
 
     writeValue(model:any):void {
-        if (!this.model || !this.model.length) {
-            this.model = {
-                countryID: 1,
-                countryName: 'United States'
-            };
-        } else {
-            this.model = model;
-        }
+        setTimeout(() => {
+            if (!model) {
+                this.model = {
+                    countryID: 1,
+                    countryName: 'United States'
+                };
+            } else {
+                model.countryName = findByCountryId(model.countryID).name;
+                this.model = model;
+                let modelState = this.states.find(state => {
```

Comment:
The current state of `getStates` only returns the `name` of the state, not the `code`. 

```
export function getStates(name) {
    let country = COUNTRIES.find(c => c.name === name.trim());
    return country ? country.states.map(s => s.name) : [];
}
```